sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -353,4 +353,8 @@ object AggUtils {
 
     finalAndCompleteAggregate :: Nil
   }
+
+  def areAggExpressionsPartial(modes: Seq[AggregateMode]): Boolean = {
+    modes.nonEmpty && modes.forall(_ == Partial)
```

Comment:
We canno apply this optimization if the empty case? e.g., 
```
scala> sql("select k from t group by k").explain()
== Physical Plan ==
*(2) HashAggregate(keys=[k#29], functions=[])
+- Exchange hashpartitioning(k#29, 200), true, [id=#47]
   +- *(1) HashAggregate(keys=[k#29], functions=[])
      +- *(1) ColumnarToRow
         +- FileScan parquet default.t[k#29] Batched: true, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex[file:/Users/maropu/Repositories/spark/spark-master/spark-warehouse/t], PartitionFilters: [], PushedFilters: [], ReadSchema: struct<k:int>
```

Reply:
I was not aware of this case, Removed the check and made it inline in its callers